ethvm/main.cpp

```diff
@@ -181,29 +178,35 @@ int main(int argc, char** argv)
 			mode = Mode::OutputOnly;
 		else if (arg == "trace")
 			mode = Mode::Trace;
-		else
-			incoming = arg;
+		else if (arg == "--input" && i + 1 < argc)
+			transactionDataString = argv[++i];
+		else if (arg == "--code" && i + 1 < argc)
+			contractCodeString = argv[++i];
 	}
 
 	VMFactory::setKind(vmKind);
 
-	bytes code;
-	if (incoming == "--" || incoming.empty())
```

Comment:
so if --call and --input are not set we assume that input data for transaction is passed either by filepath or cin input?